src/gordon_gcp/plugins/enricher.py

```diff
@@ -57,37 +59,52 @@ class GCEEnricherBuilder:
         kwargs (dict): Additional keyword arguments to pass to the
             enricher.
     """
+    VALIDATORS = [
+        gordon_gcp.enricher.GCEEnricherBuilder._validate_keyfile,
+        gordon_gcp.enricher.GCEEnricherBuilder._validate_dns_zone,
+        gordon_gcp.enricher.GCEEnricherBuilder._validate_retries
+    ]
+
     def __init__(self, config, success_channel, error_channel, **kwargs):
         self.config = config
         self.success_channel = success_channel
         self.error_channel = error_channel
         self.kwargs = kwargs
 
+    @staticmethod
+    def _validate_keyfile(config):
+        msg = []
+        if not config.get('keyfile'):
+            msg.append('The path to a Service Account JSON keyfile is required '
+                       'to authenticate to the GCE API.')
+        return msg
+
+    @staticmethod
+    def _validate_dns_zone(config):
+        msg = []
+        if not config.get('dns_zone'):
+            msg.append = 'A dns zone is required to build correct A records.'
+        if not config.get('dns_zone', '').endswith('.'):
+            msg.append = ('A dns zone must be a FQDN and end with the root '
+                          'zone (".").')
+        return msg
+
+    @staticmethod
+    def _validate_retries(config):
```

Comment:
Could these methods be abstracted into something like `_validate_config(key, err_msg, default=None)` that handles the logging and error raising? That could then be abstracted out since it would be useful to other plugins. I think it's fine to error out on the first missing key, if that is the side effect.

Reply:
Hmmmmm.  That's not a bad idea.  It also solves the pathing problem in VALIDATORS.

Reply:
I'd personally be fine with skipping it. If you'd like to keep it, consider it an edge case and just do the check after calling the original validate method. It requires a little code duplication, but still less than it is now.